tests/unit/amo/pages/TestAddon.js

```diff
@@ -1612,10 +1603,56 @@ describe(__filename, () => {
     const { store } = dispatchClientMetadata({ clientApp });
     const guid = 'this_is@a.guid';
     const addon = { ...fakeAddon, guid };
-    store.dispatch(_loadAddon({ addon }));
+    const params = { slug: guid };
+    store.dispatch(_loadAddon({ addon, slug: params.slug }));
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    renderComponent({ params, store });
+
+    sinon.assert.calledWith(
+      fakeDispatch,
+      sendServerRedirect({
+        status: 301,
+        url: `/en-US/${clientApp}${getAddonURL(addon.slug)}`,
+      }),
+    );
+    sinon.assert.callCount(fakeDispatch, 1);
+  });
+
+  it(`dispatches a server redirect when slug contains very similar characters`, () => {
+    const clientApp = CLIENT_APP_FIREFOX;
+    const { store } = dispatchClientMetadata({ clientApp });
+    const addon = { ...fakeAddon, slug: 'some-slug' };
+    // We change the slug to simulate an API response for a slug that isn't
+    // strictly the add-on's slug.
+    const params = { slug: 'söme-slug' };
+    store.dispatch(_loadAddon({ addon, slug: params.slug }));
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    renderComponent({ params, store });
+
+    sinon.assert.calledWith(
+      fakeDispatch,
+      sendServerRedirect({
+        status: 301,
+        url: `/en-US/${clientApp}${getAddonURL(addon.slug)}`,
+      }),
+    );
+    sinon.assert.callCount(fakeDispatch, 1);
+  });
+
+  it(`dispatches a server redirect when slug is the add-on's GUID using a different case`, () => {
```

Comment:
This test case seems unnecessary. We've already tested with a GUID, and a GUID with a different case shouldn't be treated any differently, afaict.